mm/ktsan/sync_atomic.c

```diff
@@ -295,38 +356,40 @@ s16 kt_atomic16_xadd(kt_thr_t *thr, uptr_t pc, uptr_t addr, s16 value)
 	s16 rv;
 
 	KT_ATOMIC_OP(rv = kt_atomic16_xadd_no_ktsan((void *)addr, value),
-			addr, kt_memory_order_acq_rel);
+		addr, kt_memory_order_acq_rel, kt_memory_order_relaxed);
 
 	return rv;
 }
 
-/* FIXME(xairy): doesn't imply memory barrier. */
 void kt_bitop_set_bit(kt_thr_t *thr, uptr_t pc, uptr_t addr, long nr)
 {
-	KT_ATOMIC_OP(kt_bitop_set_bit_no_ktsan((void *)addr, nr),
-			addr + nr / 8, kt_memory_order_release);
+	KT_ATOMIC_OP(
+		kt_bitop_set_bit_no_ktsan((void *)addr, nr), addr + nr / 8,
+		kt_memory_order_relaxed, kt_memory_order_release);
 }
 
-/* FIXME(xairy): doesn't imply memory barrier. */
 void kt_bitop_clear_bit(kt_thr_t *thr, uptr_t pc, uptr_t addr, long nr)
 {
-	KT_ATOMIC_OP(kt_bitop_clear_bit_no_ktsan((void *)addr, nr),
-			addr + nr / 8, kt_memory_order_release);
+	KT_ATOMIC_OP(
+		kt_bitop_clear_bit_no_ktsan((void *)addr, nr), addr + nr / 8,
+		kt_memory_order_relaxed, kt_memory_order_release);
 }
 
-/* FIXME(xairy): doesn't imply memory barrier. */
 void kt_bitop_change_bit(kt_thr_t *thr, uptr_t pc, uptr_t addr, long nr)
 {
-	KT_ATOMIC_OP(kt_bitop_change_bit_no_ktsan((void *)addr, nr),
-			addr + nr / 9, kt_memory_order_release);
+	KT_ATOMIC_OP(
+		kt_bitop_change_bit_no_ktsan((void *)addr, nr), addr + nr / 8,
+		kt_memory_order_relaxed, kt_memory_order_acq_rel);
 }
 
 int kt_bitop_test_and_set_bit(kt_thr_t *thr, uptr_t pc, uptr_t addr, long nr)
 {
 	int rv;
 
-	KT_ATOMIC_OP(rv = kt_bitop_test_and_set_bit_no_ktsan((void *)addr, nr),
-			addr + nr / 8, kt_memory_order_acq_rel);
+	KT_ATOMIC_OP(
+		rv = kt_bitop_test_and_set_bit_no_ktsan((void *)addr, nr),
+		addr + nr / 8,
```

Comment:
To sync on the byte the bit of which is being accessed. nr / 8 is the index of that byte.